Sources/SwiftSyntaxMacroExpansion/MacroArgument.swift

```diff
@@ -0,0 +1,138 @@
+import SwiftDiagnostics
+import SwiftSyntax
+
+enum DeclReferenceError: DiagnosticMessage {
+  case nonLiteral
+
+  var message: String {
+    switch self {
+    case .nonLiteral:
+      return "only literals are permitted"
+    }
+  }
+
+  var diagnosticID: MessageID {
+    .init(domain: "SwiftMacros", id: "\(self)")
+  }
+
+  var severity: DiagnosticSeverity {
+    .error
+  }
+}
+
+class CheckDeclReferenceVisitor: SyntaxAnyVisitor {
```

Comment:
I don’t think this name is quite accurate. I would prefer something like `OnlyLiteralExprChecker` or something of the kind because it also rejects eg. `{ return 1 }()`, which doesn’t contain any reference. 

Also a short doc comment would be appreciated.